components/Main/ContactUs/MessageForm/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react'
+import style from './MessageForm.module.css'
+import type { ChangeEvent, Dispatch, FormEvent, SetStateAction } from 'react'
+
+export type ContactUsFormProps = {
+  setIsSuccessful: Dispatch<SetStateAction<boolean>>
+}
+
+export default function MessageForm({ setIsSuccessful }: ContactUsFormProps): JSX.Element {
+  const NAME = 'name'
+  const EMAIL = 'email'
+  const BODY = 'body'
+  const MAX_BODY_LENGTH = 500
+
+  const [message, setMesage] = useState({
+    [NAME]: '',
+    [EMAIL]: '',
+    [BODY]: '',
+    bodyLength: 0,
+  })
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    event.preventDefault()
+    if (event.target.name === BODY) {
+      setMesage({
+        ...message,
+        [event.target.name]: event.target.value,
+        bodyLength: event.target.value.length,
```

Comment:
Forms are always a fun one. Is the `bodyLength` relevant to the rest of the state or can it be managed separately?